### ISSUE TITLE
Expected a Color Instance

### ISSUE THREAD

**Issue by tresf** [qzind/tray#276]
[This PDF](https://github.com/qzind/tray/files/1705990/SC5361044390.pdf) will crash PDF printing in QZ Tray with `rasterize: false`.

* This bug does NOT occur if printing using pdfbox 2.0.8 command line.
* This bug does NOT occur if printing using `rasterize: true`
* This bug does NOT occur on Mac
* This bug does NOT occur on the 2.1 branch

```
[ERROR] 2018-02-08 01:56:37,352 @ qz.utils.PrintingUtilities:?
        Failed to print
java.lang.IllegalArgumentException: Expected a Color instance
        at sun.print.PathGraphics.fill(Unknown Source)
        at org.apache.pdfbox.rendering.PageDrawer.fillPath(PageDrawer.java:719)
        at org.apache.pdfbox.contentstream.operator.graphics.FillNonZeroRule.pro
cess(FillNonZeroRule.java:36)
        at org.apache.pdfbox.contentstream.PDFStreamEngine.processOperator(PDFSt
reamEngine.java:838)
        at org.apache.pdfbox.contentstream.PDFStreamEngine.processStreamOperator
s(PDFStreamEngine.java:495)
        at org.apache.pdfbox.contentstream.PDFStreamEngine.processStream(PDFStre
amEngine.java:469)
        at org.apache.pdfbox.contentstream.PDFStreamEngine.processPage(PDFStream
Engine.java:150)
        at org.apache.pdfbox.rendering.PageDrawer.drawPage(PageDrawer.java:203)
        at org.apache.pdfbox.rendering.PDFRenderer.renderPageToGraphics(PDFRende
rer.java:184)
        at org.apache.pdfbox.printing.PDFPrintable.print(PDFPrintable.java:190)
        at qz.printer.PDFWrapper.print(Unknown Source)
        at qz.printer.BookBundle.print(Unknown Source)
        at sun.print.RasterPrinterJob.printPage(Unknown Source)
        at sun.print.RasterPrinterJob.print(Unknown Source)
        at qz.printer.action.PrintPixel.printCopies(Unknown Source)
        at qz.printer.action.PrintPDF.print(Unknown Source)
        at qz.utils.PrintingUtilities.processPrintRequest(Unknown Source)
        at qz.ws.PrintSocketClient.processMessage(Unknown Source)
        at qz.ws.PrintSocketClient.onMessage(Unknown Source)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.lang.reflect.Method.invoke(Unknown Source)
        at org.eclipse.jetty.websocket.common.events.annotated.CallableMethod.ca
ll(CallableMethod.java:70)
        at org.eclipse.jetty.websocket.common.events.annotated.OptionalSessionCa
llableMethod.call(OptionalSessionCallableMethod.java:68)
        at org.eclipse.jetty.websocket.common.events.JettyAnnotatedEventDriver$2
.run(JettyAnnotatedEventDriver.java:210)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPoo
l.java:635)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool
.java:555)
        at java.lang.Thread.run(Unknown Source)
[TRACE] 2018-02-08 01:56:37,352 @ qz.utils.PrintingUtilities:?
        Returning processor back to pool
```

**Comment by bberenz** [comment]
Fix under 9ed2203231102418c7c2f88a6dcf106bd50521ce
Looks like the short-circuit can cause issues? We never really profiled the code to know if it reduces memory footprint, so I've just removed it to run before we added it since it can do this.

**Comment by tresf** [comment]
Confirmed.  This fix will be available in 2.0.6.
